core/unit_test/headers_self_contained/CMakeLists.txt

```diff
@@ -17,3 +17,4 @@ foreach (_header ${KOKKOS_CORE_HEADERS} ${KOKKOS_CONTAINERS_HEADERS} ${KOKKOS_AL
   target_link_libraries(${header_test_name} PRIVATE Kokkos::kokkos)
```

Comment:
I guess you could use `KOKKOS_ADD_EXECUTABLE` here instead. Can you try that?
On the other hand, I don't have a problem with disabling that test for `Trilinos`.

Reply:
@ndellingwood I would prefer the resolution suggested by Daniel here.

Reply:
I'll come back to this soon once I finish debugging in Trilinos

Reply:
I tried making the `KOKKOS_ADD_EXECUTABLE` change, but I'm not sure how to handle the compile definitions in the `target_compile_defines` line for TriBITS cases.

The TriBITS documentation from https://tribits.org/doc/TribitsDevelopersGuide.html indicates that `TARGET_DEFINES -D<define0>` will add the listed compile defines to `target_compile_defines`

I tried adding a few lines to `kokkos/cmake/kokkos_tribits.cmake` to recognize `TARGET_DEFINES` and set the defines:

```cmake
--- a/cmake/kokkos_tribits.cmake
+++ b/cmake/kokkos_tribits.cmake
@@ -119,7 +119,7 @@ FUNCTION(KOKKOS_ADD_EXECUTABLE ROOT_NAME)
     CMAKE_PARSE_ARGUMENTS(PARSE
       "TESTONLY"
       ""
-      "SOURCES;TESTONLYLIBS"
+      "SOURCES;TESTONLYLIBS;TARGET_DEFINES"
       ${ARGN})
 
     SET(EXE_NAME ${PACKAGE_NAME}_${ROOT_NAME})
@@ -131,6 +131,10 @@ FUNCTION(KOKKOS_ADD_EXECUTABLE ROOT_NAME)
     #All executables must link to all the kokkos targets
     #This is just private linkage because exe is final
     TARGET_LINK_LIBRARIES(${EXE_NAME} PRIVATE Kokkos::kokkos)
+    IF (PARSE_TARGET_DEFINES)
+      MESSAGE(STATUS "TESTING PARSE_TARGET_DEFINES: ${PARSE_TARGET_DEFINES}")
+      TARGET_COMPILE_DEFINITIONS(${EXE_NAME} PRIVATE ${PARSE_TARGET_DEFINES})
+    ENDIF()
   endif()
 ENDFUNCTION()
```

I replaced the existing `add_executable`, `target_link_libraries`, and `target_compile_definitions` in `kokkos/core/unit_test/headers_self_contained/CMakeLists.txt` with the following:
```cmake
  KOKKOS_ADD_EXECUTABLE(
    ${header_test_name}
    SOURCES tstHeader.cpp
    TARGET_DEFINES -DPRIVATE KOKKOS_HEADER_TEST_NAME=${_header}
  )
```

but I get compilation errors in Trilinos like
```bash
[  2%] Building CXX object kokkos/core/unit_test/headers_self_contained/CMakeFiles/KokkosCore_Kokkos_HeaderSelfContained_Kokkos_Future_hpp.dir/tstHeader.cpp.o
/ascldap/users/ndellin/Trilinos/kokkos/core/unit_test/headers_self_contained/tstHeader.cpp:8:10: fatal error: Kokkos_Random.hpp: No such file or directory
 #include KOKKOS_HEADER_TO_TEST
...
```

not sure what to change to do this properly, I must not be using `TARGET_DEFINES` properly with TriBITS? The changes worked fine with stand-alone kokkos (configure and build successful).
I'll need some help from more qualified CMake and TriBITS folks to make any further progress, unless we want to stick with the current state of the PR, @jjwilke @dalg24 @masterleinad

Reply:
This all looks good - except: isn't it `-DKOKKOS_HEADER_TO_TEST` ?

Reply:
Slack-chatted with @jjwilke and he figured out the issue: In the Trilinos build, the failures are occurring with files from `containers` and `algorithms` (i.e. not the core subpackage). TriBITS doesn't know anything about the dependence, and because we're in the core subpackage it's not including container or algorithms header paths.
@dalg24 @crtrott to minimize delays in release testing, can we go with the current PR's workaround and aim for a proper fix in the next release?